### PR TITLE
Feature/web 404 redirect rework

### PR DESCRIPTION
**Pull request by stefanhar**
# \<Description\>

https://app.asana.com/0/1199123945323117/1199603312595100

## What

- Make 404 redirect support all linkable types by using reference link type
- Make link resolution be more standardized
- Fixed news page not returning 404

## Why

- This allows for simpler resolution of links

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/web/hooks/useLinkResolver/useLinkResolver.tsx [hunk 1]
```diff
@@ -103,6 +104,20 @@ const convertToRegex = (routeTemplate: string) =>
...
+  path: string,
+  template: string,
+): string[] => {
+  const pathParts = path.split('/')
+  const templateParts = template.split('/')
+
+  return pathParts.filter((_, index) => {
+    return templateParts[index]?.startsWith('[') ?? ''
+  })
+}
```
Comment (siggakogk) [inline apps/web/hooks/useLinkResolver/useLinkResolver.tsx:120]: add unit tests?

File: apps/web/hooks/useLinkResolver/useLinkResolver.tsx [hunk 2]
```diff
@@ -178,9 +193,18 @@ export const typeResolver = (
...
+      if (
+        (!skipDynamic && path?.match(regex)) ||
+        (skipDynamic && path?.startsWith(routeTemplate))
+      ) {
+        console.log('Found type!', type, locale)
+        return {
+          slug: extractSlugsByRouteTemplate(path, routeTemplate),
+          type,
+          locale,
+        } as TypeResolverResponse
```
Comment (siggakogk) [inline apps/web/hooks/useLinkResolver/useLinkResolver.tsx:207]: is this method tested? do you need to add more here or update the tests to consider this case?

File: apps/web/hooks/useLinkResolver/useLinkResolver.tsx (hunk 1 above)
Comment (stefanhar) [reply apps/web/hooks/useLinkResolver/useLinkResolver.tsx]: Yes

File: apps/web/hooks/useLinkResolver/useLinkResolver.tsx (hunk 2 above)
Comment (stefanhar) [reply apps/web/hooks/useLinkResolver/useLinkResolver.tsx]: Yes, this is tested via typeResolver test but will add tests specific to this

**Review by siggakogk — APPROVED** [review]
lgtm!

**Pull request merged**